src/contexts/product-context.js

```diff
@@ -0,0 +1,24 @@
+import { createContext, useContext, useEffect, useState } from "react";
+import axios from "axios";
+
+const ProductContext = createContext(null);
+export const useProduct = () => useContext(ProductContext);
+
+export const ProductProvider = ({ children }) => {
+  const [productData, setProductData] = useState([]);
+  useEffect(() => {
+    (async () => {
+      try {
+        let res = await axios.get("/api/products");
+        setProductData(res.data.products);
+      } catch (error) {
+        console.log(error.message);
```

Comment:
suggestion: you can use console.error(), so error would be displayed in error section separately in the console of browser dev tool.